### PR TITLE
fix(startWith): accepts N arguments and returns correct type

### PR DESCRIPTION
**Pull request by benlesh**
- Improves documentation a bit
- Updates type tests

NOTE: Calls with more than 7 arguments that have a Scheduler at the end will return the wrong time. This is consider a corner case and low-risk

BREAKING CHANGE: `startWith` will return incorrect types when called with more than 7 arguments and a scheduler. Passing scheduler to startWith is deprecated

### CODE REVIEW

**Comment by cartant** [comment]
Note to self: look at https://github.com/ReactiveX/rxjs/pull/5157 when reviewing this.

**Comment by benlesh** [comment]
ping @cartant

File: spec-dtslint/operators/startWith-spec.ts [hunk 1]
```diff
@@ -1,30 +1,24 @@
...
+  const r1 = of(a).pipe(startWith(b)); // $ExpectType Observable<A | B>
+  const r2 = of(a).pipe(startWith(b, c)); // $ExpectType Observable<A | B | C>
+  const r3 = of(a).pipe(startWith(b, c, d)); // $ExpectType Observable<A | B | C | D>
+  const r4 = of(a).pipe(startWith(b, c, d, e)); // $ExpectType Observable<A | B | C | D | E>
+  const r5 = of(a).pipe(startWith(b, c, d, e, f)); // $ExpectType Observable<A | B | C | D | E | F>
+  const r6 = of(a).pipe(startWith(b, c, d, e, f, g)); // $ExpectType Observable<A | B | C | D | E | F | G>
+  const r7 = of(a).pipe(startWith(b, c, d, e, f, g, h)); // $ExpectType Observable<A | B | C | D | E | F | G | H>
 });
 
 it('should infer correctly with only a scheduler', () => {
```
Comment (cartant) [inline spec-dtslint/operators/startWith-spec.ts]: Minor nit: test description is now out-of-date

File: src/internal/operators/startWith.ts [hunk 2]
```diff
@@ -41,37 +35,37 @@ export function startWith<T, D = T>(...array: Array<D | SchedulerLike>): Operato
...
- * @owner Observable
+ * @param values - Items you want the modified Observable to emit first.
+ *
+ * @see endWith
+ * @see finalize
+ * @see concat
  */
-export function startWith<T, D>(...array: Array<T | SchedulerLike>): OperatorFunction<T, T | D> {
-  const scheduler = array[array.length - 1] as SchedulerLike;
+export function startWith<T, D>(...values: Array<T | SchedulerLike>): OperatorFunction<T, T | D> {
```
Comment (cartant) [inline src/internal/operators/startWith.ts]: I see that https://github.com/ReactiveX/rxjs/pull/5157 has been merged, so I'm guessing the change in that PR will be picked up once this is rebased.
